android/src/main/java/com/reactnativerowndplugin/RowndPluginModule.kt

```diff
@@ -112,13 +112,19 @@ class RowndPluginModule(reactContext: ReactApplicationContext) : ReactContextBas
       }
     }
 
-//    @ReactMethod
-//    suspend fun getAccessToken(promise: Promise) {
-//      try {
-//        val accessToken = Rownd.getAccessToken()
-//        promise.resolve(accessToken ?: "")
-//      } catch (e: Throwable) {
-//        promise.reject("GET_ACCESS_TOKEN_ERROR: ", e)
-//      }
-//    }
+    @ReactMethod
+    fun getAccessToken(promise: Promise) {
+      coroutineScope = CoroutineScope(Dispatchers.IO).launch {
+        try {
+          val accessToken = Rownd.getAccessToken()
+          accessToken?.let {
+            promise.resolve(accessToken)
+          } ?: run {
+            promise.reject("ROWND PLUGIN MODULE ERROR: NO ACCESS TOKEN")
```

Comment:
What happens to the caller in this case? On native, we just return `null/nil` if the access token is unavailable.

Reply:
For async getAccessToken, it gets caught by the .catch instead of .then

Would you want me to just return a null/empty string instead on the JS side, instead of throwing an error?

Reply:
Updated so that it returns an empty string if Rownd.getAccessToken is null